### PR TITLE
Include add_before and add_after in new item functions

### LINKED ISSUE #66: Auto add Files
Add to `vt_use_req()`, `vt_use_test_case()`, and `vt_use_test_code()` the ability to automatically be added to validation.yml. 

Will need default add methodology too: group reqs, test case and test code?

To avoid having to do:

```r
vt_use_req("New_requirement_100")
vt_add_file_to_config("New_requirement_100.md", after = "New_requirement_099.md")
```

More like:
```r
vt_use_req("New_requirement_100", add_after = "New_requirement_099.md")
```

### PR DESCRIPTION
**Pull request by elimillera**
This PR adds the `add_file_to_config` functionality and adds the parameters `add_before` and `add_after` to be passed into the `before` and `after` parameters of `add_file_to_config`.

I'm not sure what the plans are for merging things in but I can start working off this branch to add 'tidy_select' functionality if I'm on the right track.

Closes #66

### CODE REVIEW

File: R/code.R [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # The rest of this documentation is in case.R
 #' @export
 #' @rdname new_item
-vt_use_test_code <- function(name, username = vt_username(), open = interactive()) {
+vt_use_test_code <- function(name, username = vt_username(), open = interactive(), pkg = ".",
```
Comment (thebioengineer) [inline R/code.R:4]: In the develop version of the package, `pkg` for all the vt_use_* functions is no longer necessary. you can remove that argument from these functions and from `vt_add_file_to_config()`.

File: R/code.R [hunk 2]
```diff
@@ -21,6 +22,10 @@ vt_use_test_code <- function(name, username = vt_username(), open = interactive(
                     ))
   }
 
+  # Add file to validation configuration
+  vt_add_file_to_config(filename = name, pkg = pkg, after = add_after,
```
Comment (thebioengineer) [inline R/code.R:26]: `pkg` is no longer an argument in `vt_add_file_to_config()`, which is probably why the test is failing CI.

File: R/case.R [hunk 3]
```diff
@@ -5,6 +5,10 @@
 #'   are located at `./inst/validation/<ItemType>/{name}`.
 #' @param username The username to insert into the validation item as the author.
 #' @param open Should the newly made file be opened for editing.
+#' @param pkg Top-level of package
```
Comment (thebioengineer) [inline R/case.R:8]: remove this line

**Review by thebioengineer — CHANGES_REQUESTED** [review]
Good with the concept, but if you could go back and clean up the `pkg` argument, I think this PR will pass CI and be good to merge in. Hopefully that is quick enough!

**Comment by thebioengineer** [comment]
there are tests failing now that were explicitly adding the files into the validation.yml! haha. at least we know if works!!

**Comment by elimillera** [comment]
Yep! I'm working on those now. Should be ready by end of today

**Comment by elimillera** [comment]
Hey @thebioengineer, I had some trouble with the tests. I think my VM must have a weird pdf render because I get a lot of errors on my machine. But they are passing here so it should be good to merge for now.

**Comment by mariev** [comment]
@elimillera did pulling branch `develop` address the pdf build errors on your side? we are temporarily blocked from gh-actions due to https://github.com/r-lib/actions/issues/280 but no errors when I run devtools::check locally

**Review by mariev — APPROVED** [review]

**Pull request merged**